### ISSUE TITLE
Conversion from Binding[Iterable[A]] to BindingSeq[A]

### ISSUE THREAD

**Issue by Axiometry** [ThoughtWorksInc/Binding.scala#299]
You can hypothetically convert a BindingSeq[A] into a Binding[Iterable[A]] with the `all` def (though as per #294 it doesn't seem to be working in Scala 2.13 right now).

Is there a supported conversion in the other direction? I came up with this hack ([ScalaFiddle](https://scalafiddle.io/sf/YafJxJX/3)):

```scala
def bindingSeq[A](seqInBinding: Binding[Iterable[A]]): BindingSeq[A] =
  Constants(1).flatMap { _ =>
    Constants(seqInBinding.bind.toSeq: _*)
  }
```

Is there a more idiomatic method? Also, this hack doesn't seem to work when positioned in an implicit value class ([ScalaFiddle](https://scalafiddle.io/sf/9P9jCLd/2)):

```scala
implicit class Wrapper[A](val seqInBinding: Binding[Iterable[A]]) extends AnyVal {
  def bindingSeq: BindingSeq[A] =
    Constants(1).flatMap { _ =>
      Constants(seqInBinding.bind.toSeq: _*)
    }
}
```

`Could not find proxy for   x$1: scala.this.Int in List(value x$1, value $anonfun, method bindingSeq, class Wrapper, module class Experiment, module class ScalaFiddle, package class , package class ) (currentOwner= value $anonfun `

(I repeated all these experiments on org.lrng.binding.html 1.0.3+9-7d1ef722 with scala 2.13.1; same results as in the ScalaFiddles.)

**Comment by Axiometry** [comment]
I've come up with a slightly more elegant solution:
```scala
def bindingSeq[A](seqInBinding: Binding[Iterable[A]]): BindingSeq[A] =
  Constants(Binding(Constants(seqInBinding.bind.toSeq: _*))).flatMap(_.bind)
```
(and for related conversions:)
```scala
def bindingSeq[A](seqOfBinding: Iterable[Binding[A]]): Binding[Iterable[A]] =
  Constants(seqOfBinding.toSeq: _*).map(_.bind)
def bindingSeq[A](seqOfBinding: Iterable[Binding[A]]): BindingSeq[A] =
  Constants(seqOfBinding.toSeq: _*).map(_.bind)
```
Feel free to close this if employing `Constants` in this way is the most idiomatic method.

Note that this still doesn't work when positioned in an implicit value class:
```scala
implicit class Wrapper[A](val seqInBinding: Binding[Iterable[A]]) extends AnyVal {
  def bindingSeq: BindingSeq[A] =
    Constants(Binding(Constants(seqInBinding.bind.toSeq: _*))).flatMap(_.bind)
}
```
`Could not find proxy for    x$4: binding.this.Binding in List(value x$4, value $anonfun, method bindingSeq, class Wrapper, module class ScalaFiddle, package class , package class ) (currentOwner= value $anonfun )`

**Comment by Axiometry** [comment]
I've created a separate issue for the implicit value class error: #300

**Comment by Atry** [comment]
> Could not find proxy for ...

This is definitely a Scala compiler bug

**Comment by Atry** [comment]
`vars.all.bind ` was fixed in #302
